### DISCUSSION TITLE
isDirty becoming true when a form is rendered again.

### DISCUSSION THREAD

**Discussion by adriantrunzo** [react-hook-form/react-hook-form#4489] [Q&A]
Hi,

I was trying to troubleshoot the `isDirty` property on a form I am working on because `isDirty` always seemed to be `true` to start. In other words, in my particular case, even though the form was untouched and I was providing `defaultValues`, `isDirty` was `true`. I have figured out why this behavior was occurring and was able to reproduce it in a sandbox: https://codesandbox.io/s/is-dirty-rendering-quirk-g1s24?file=/src/App.js

Basically, if the component renders the form, then renders something else, and finally renders the form again, `isDirty` is `true`, even though no inputs on the form are changed by the user. In my particular case, I have solved the issue by adding the appropriate logic to avoid the initial rendering of the form. My question here is: why should the rendering pattern demonstrated in the codesandbox cause `isDirty` to become `true`?
